apps/client/app/purchase-orders/[id]/components/detail.tsx

```diff
@@ -36,6 +47,17 @@ export default function Detail({purchaseOrder}: Props) {
               {term: 'Line Item Count', value: purchaseOrder.lineItems.length},
             ]}
           />
+          {hasPermission('approve', 'purchase_order') &&
+            purchaseOrder.status === 'pending' && (
+              <div className="card-actions">
+                <button className="btn btn-primary" onClick={() => onApprove()}>
+                  Approve
+                </button>
+                <button className="btn btn-danger" onClick={() => onDeny()}>
+                  Deny
+                </button>
+              </div>
+            )}
```

Comment:
Shows the approval actions if this user has approval permissions and the order is pending.